roslibrust/src/ros1/node.rs

```diff
@@ -372,6 +346,52 @@ impl Node {
             }
         }
     }
+
+    async fn register_publisher(
+        &mut self,
+        topic: String,
+        topic_type: &str,
+        queue_size: usize,
+        msg_definition: String,
+        md5sum: String,
+    ) -> Result<mpsc::Sender<Vec<u8>>, Box<dyn std::error::Error>> {
+        if let Some(handle) = self.publishers.iter().find_map(|entry| {
+            if entry.key().as_str() == &topic {
+                if entry.topic_type() == topic_type {
+                    Some(Ok(entry.get_sender()))
+                } else {
+                    Some(Err(Box::new(std::io::Error::from(
```

Comment:
This error type isn't unreasonable, but I think it's very not human readable.

Fine to merge, but would prefer we give an error that clearly says "Attempted to register a publisher of a different type that was already registered to {topic}, new type, existing type"